prospector/formatters/pylint.py

```diff
@@ -32,15 +31,20 @@ def render(self, summary: bool = True, messages: bool = True, profile: bool = Fa
             output.append(
                 template
                 % {
-                    "path": self._make_path(message.location.path),
+                    "path": self._make_path(message.location),
                     "line": message.location.line,
                     "source": message.source,
                     "code": message.code,
                     "function": message.location.function,
                     "message": message.message.strip(),
                 }
             )
+        return output
 
+    def render(self, summary: bool = True, messages: bool = True, profile: bool = False) -> str:
+        output: list[str] = []
+        if messages:
+            output.extend(self.render_messages())
```

Comment:
This nullify all the argument forcing the default, is  this really  what we want to do ? We can't use formatter interchangeably if we do that. (Breaking the L in SOLID)

Reply:
> This nullify all the argument forcing the default

I don't understand what you want to say with that?

> We can't use formatter interchangeably if we do that.

To be able to use the formatter interchangeable is my goal, then we have a misunderstanding somewhere!

In fact, it should be possible to move this method to base_summary.py.

Reply:
I meant previously we could give " summary: bool = True, messages: bool = True, profile: bool = False" to render. I realize now that is wasn't used at all so the behavior doesn't change.